runtime/development/src/lib.rs

```diff
@@ -1325,38 +1329,74 @@ impl xcm_executor::Config for XcmConfig {
 /// We need to ensure we have at least one rule per token we want to handle or else
 /// the xcm executor won't know how to charge fees for a transfer of said token.
 pub type Trader = (
-	FixedRateOfFungible<NativePerSecond, ()>,
-	FixedRateOfFungible<UsdPerSecond2000, ()>,
-	FixedRateOfFungible<UsdPerSecond3000, ()>,
+	FixedRateOfFungible<KsmPerSecond, ToTreasury>,
+	FixedRateOfFungible<NativePerSecond, ToTreasury>,
+	FixedRateOfFungible<KUsdPerSecond, ToTreasury>,
+	FixedRateOfFungible<UsdPerSecond, ToTreasury>,
+	FixedRateOfFungible<UsdPerSecondSibling, ToTreasury>,
 );
 
+pub struct ToTreasury;
+impl TakeRevenue for ToTreasury {
+	fn take_revenue(revenue: MultiAsset) {
+		use orml_traits::MultiCurrency;
+		use xcm_executor::traits::Convert;
+
+		if let MultiAsset {
+			id: Concrete(location),
+			fun: Fungible(amount),
+		} = revenue
+		{
+			if let Ok(currency_id) =
+				<CurrencyIdConvert as Convert<MultiLocation, CurrencyId>>::convert(location)
+			{
+				let _ = OrmlTokens::deposit(currency_id, &TreasuryAccount::get(), amount);
+			}
+		}
```

Comment:
Where is `take_revenue` called and what happens if the given `CurrencyId`-`Multilocation`-combination is not convertible?

Reply:
`ToTreasury` is called by the `Trader` rules. Is the pair is not convertible, it should have failed before reaching that point, since that would mean that this chain didn't know about that token neither how to charge fees for it. Let me know if that makes sense!